lib/endpoint-with-defaults.ts

```diff
@@ -0,0 +1,8 @@
+export = endpointWithDefaults
+
+import merge = require('./merge')
+import parse = require('./parse')
+
+function endpointWithDefaults (defaults: typeof import("./defaults"), route: string, options: any) {
```

Comment:
the arguments are either `route: string, options: any` or just `options: any`.

And I wonder if there is something better than `any`? If only `options` are passed, the object must contain `method` and `url`, if both `route` and `options` are set than `options` should not include `method` or `url`

Reply:
So, `route` and `options` should be optional?

Reply:
I think it would make sense to use parameter overrides

```typescript
type Route = string
type Options = {
    headers?: Object,
    request?: Object,
    baseUrl?: string,
    data?: any
    [option: string]: any
}
type RouteOptions = {
    method: Method,
    url: KnownPath
}
type RequestOptions = {
    method: Method,
    url: string,
    headers: object,
    body?: any,
    request?: object
}

function endpointWithDefaults(defaults: Options, route: Route, options: Options): RequestOptions;
function endpointWithDefaults(defaults: Options, options: RouteOptions & Options): RequestOptions;
```

Something like this? I took that from https://github.com/octokit/endpoint.js/issues/23

Reply:
Unfortunately that will not work. TypeScript complains since the functions don't have a similar signature